### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by oovillagran**
<h1>Hey there 👋</h1>

In this branch, I set up the User, Post, Comment, and Like models. I also, use the CRUD methods, and finally custom methods were created. I put the CRUD and custom methods in the db/seeds.rb file.

Here are the *project requirements*:

**Create models**
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
- Set up associations between models.
     - Remember that ```author_id``` column in ```posts``` table should be the foreign_key for the ```users``` table.

**Use models to insert data**
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
```
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
```
- Create at least 4 posts written by one of the users you created by running the following code:

``` 
first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
```
- Create at least 6 posts comments for one of the posts you created by running the following code:
```
Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

**Create custom methods**
Your models should include some additional methods.

- Users
     - A method that returns the 3 most recent posts for a given user.
- Posts
     - A method that updates the posts counter for a user.
     - A method which returns the 5 most recent comments for a given post.
- Comments
     - A method that updates the comments counter for a post.
- Likes
     - A method that updates the likes counter for a post.

**You can use the command rails db:seed to test the created methods**
 
<h2>Thanks for your kind review and feedback</h2>

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
+  after_create :update_user_posts_counter
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (rloterh) [inline app/models/post.rb]: - [x] [OPTIONAL]
Following best practices `update_posts_counter` should be marked as _private_ after implementing the `after_create` method.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def five_most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_likes_counter
+    update(likes_counter: likes.count)
+  end
```
Comment (rloterh) [inline app/models/post.rb]: - [x] [OPTIONAL]
`update_like_counter` method is already defined in the _like_ model, which is required. It's therefore redundant to have a duplicate of the `update_like_counter` in the _post_ model.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post
+
+  after_create :update_post_comments_counter
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] [OPTIONAL]
Following best practices `update_comment_counter` should be marked as _private_ after implementing the `after_create` method.

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post
+
+  after_create :update_post_likes_counter
+
+  def update_post_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (rloterh) [inline app/models/like.rb]: - [x] OPTIONAL
Following best practices `update_like_counter` should be marked as _private_ after implementing the `after_create` method.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Oscar,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on establising model association!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oovillagran** [comment]
Thanks @rloterh !

**Pull request merged**